### PR TITLE
Authentication API

### PR DESCRIPTION
**Pull request by kaylaleung**
Add authentication to main index.html page to redirect to login page. Also add login verification for /blog.html?user=admin for which the blog editor will be visible. Editor will not be viewable to other non-admin users.

### CODE REVIEW

File: portfolio/src/main/webapp/blog.html [hunk 1]
```diff
@@ -55,40 +56,43 @@ <h4>Software Engineering Intern</h4>
...
-              <input type="text" id="category" name="category">
-            </div>
-            <div class="input-box">
-              <label for="post">Write Blog Post</label>
-              <textarea id="post" name="blog-input" cols ="100" rows="8"></textarea>
+          <div id="see-post">
+            <h2 id="title-container">Blog Title</h2>
+            <div id="post-container"></div>
+            <h2>All Comments</h2>
+            <div id="comment-container">
```
Comment (anbenson) [inline portfolio/src/main/webapp/blog.html]: I know this wasn't added in this PR but a div wrapping a ul element is pretty sus, if you're trying to apply styles, why not just apply them to the ul?

File: portfolio/src/main/webapp/blog.html [hunk 2]
```diff
@@ -47,6 +47,7 @@ <h4>Software Engineering Intern</h4>
             <li class="tab"><a href="/index.html#experience"><i class="fas fa-address-card"></i>Experience</a></li>
             <li class="tab"><a href="/index.html#projects"><i class="fas fa-project-diagram"></i>Projects</a></li>
             <li class="tab"><a href="/index.html#blog" class="active"><i class="fas fa-blog"></i>Blog</a></li>
+            <li class="tab"><a id="auth" href=""><i class="fas fa-user"></i>Login</a></li>
```
Comment (anbenson) [inline portfolio/src/main/webapp/blog.html:50]: [optional] From a design/HCI perspective, it's non-intuitive to the user that 4 of the 5 links scroll you down the page and one takes you away.

File: portfolio/src/main/webapp/blog.html [hunk 3]
```diff
@@ -55,40 +56,43 @@ <h4>Software Engineering Intern</h4>
...
+              <div class="input-box">
+                <label for="feedback">Comment</label>
+                <textarea id="feedback" name="text-input" cols="100" rows="8" maxlength="10000" required></textarea>
+              </div>
+              <input type="hidden" name="current-id" id="current-id" value="">
+              <input type="submit" value="Post a Comment">
+            </form>
+          </div>
+         <form id="write-post" action="/post" method="POST">
+          <h2 id="title-container">Write a Post</h2>
```
Comment (anbenson) [inline portfolio/src/main/webapp/blog.html]: This is 1 space tabbed :(

File: portfolio/src/main/webapp/blogScript.js [hunk 4]
```diff
@@ -1,14 +1,23 @@
+const role = new URL(document.URL).searchParams.get('user');
```
Comment (anbenson) [inline portfolio/src/main/webapp/blogScript.js]: Can you move these out of global variables?

File: portfolio/src/main/webapp/blogScript.js [hunk 5]
```diff
@@ -54,3 +63,12 @@ function createCommentElement(comment) {
 
   return commentElement;
 }
+
+function setAdmin(admin) {
+  fetch('/auth').then(response => response.json()).then((log) => {  
+    if ("" === log.logoutUrl || log.email !== "Kaylamyhome@gmail.com") {
```
Comment (anbenson) [inline portfolio/src/main/webapp/blogScript.js]: Can you make the first comparison not inverted e.g. `log.logoutUrl === ''`?

Also, single quotes in JS.

File: portfolio/src/main/webapp/blogScript.js [hunk 6]
```diff
@@ -1,14 +1,23 @@
...
+  const user = document.getElementById('see-post');
+  const admin = document.getElementById('write-post');
+  if (role !== "admin" && id !== null) {
+    getID();
     getPost();
     getComments();
+    user.style.display = 'block';
+  }
+  else {
+    setAdmin(admin);
```
Comment (anbenson) [inline portfolio/src/main/webapp/blogScript.js:15]: Okay, so as you probably know, this isn't security in the slightest.

A secure implementation would require checking the user service on the server for each and every sensitive request sent to it. But we'd also have talked through how to implement a secure system prior to implementation :)

File: portfolio/src/main/webapp/blogScript.js (hunk 4 above)
Comment (anbenson) [reply portfolio/src/main/webapp/blogScript.js]: Also consider reusing the search params?

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -17,3 +18,17 @@ function toggleActiveSection(tab) {
     tab.target.classList.add('active');
   }
 }
+
+function setLogin() {
+  fetch('/auth').then(response => response.json()).then((log) => {  
+    let login = document.getElementById('auth');
+    if ("" === log.logoutUrl) {
```
Comment (anbenson) [inline portfolio/src/main/webapp/script.js]: Please fix inversion

Comment (anbenson) [reply portfolio/src/main/webapp/script.js]: Also single quotes

**Review by anbenson — COMMENTED** [review]
Approved

File: portfolio/src/main/webapp/blogScript.js [hunk 8]
```diff
@@ -1,14 +1,23 @@
...
 });
 
-const id = new URL(document.URL).searchParams.get('id');
-
-function getURL() {
-    const urlElement = document.getElementById('current-id');
-    urlElement.value = id;
+function getID() {
+    const idElement = document.getElementById('current-id');
+    idElement.value = id;
```
Comment (joanchenx) [inline portfolio/src/main/webapp/blogScript.js:21]: Nit: I think these lines are double tabbed

**Review by joanchenx — COMMENTED** [review]
Sorry I didn't see this earlier! For future PRs, could you add Andrew and I as reviewers? That way, it pops up on my homepage. I try and catch PRs by going to each of your repos so no worries if you forget once in a while.

**Comment by kaylaleung** [comment]
> Sorry I didn't see this earlier! For future PRs, could you add Andrew and I as reviewers? That way, it pops up on my homepage. I try and catch PRs by going to each of your repos so no worries if you forget once in a while.

Completely my bad, I opened the PR on Saturday and I didn't want to bother the two of you over the weekend and then forgot to assign this morning!

File: portfolio/src/main/webapp/blog.html [hunk 9]
```diff
@@ -87,8 +89,8 @@ <h2>Leave a Reply</h2>
               <label for="post">Write Blog Post</label>
               <textarea id="post" name="blog-input" cols ="100" rows="8"></textarea>
             </div>
-            <input type="submit" value="Submit Blog Post">
-          </form>
+          <input type="submit" value="Submit Blog Post">
+        </form>
```
Comment (joanchenx) [inline portfolio/src/main/webapp/blog.html]: Nit: spacing, I think these should be tabbed over?

**Review by joanchenx — APPROVED** [review]
Approved, looks good!

(also re: the reviewer thing, no worries! I just wanted to make sure you aren't waiting on my review for too long :P)

**Pull request merged**